src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -480,6 +510,53 @@ private ClusterDescription ProcessShardedChange(ClusterDescription clusterDescri
             return clusterDescription.WithServerDescription(args.NewServerDescription);
         }
 
+        private ClusterDescription ProcessStandaloneChange(ClusterDescription clusterDescription, ServerDescriptionChangedEventArgs args)
+        {
+            if (args.NewServerDescription.Type != ServerType.Standalone)
+            {
+                return RemoveServer(clusterDescription, args.NewServerDescription.EndPoint, "Server is not a standalone server.");
+            }
+
+            return clusterDescription.WithServerDescription(args.NewServerDescription);
+        }
+
+        void IDnsMonitoringCluster.ProcessDnsResults(List<DnsEndPoint> dnsEndPoints, CancellationToken cancellationToken)
+        {
+            var newServers = new List<IClusterableServer>();
+            lock (_updateClusterDescriptionLock)
```

Comment:
Get the lock before starting to examine and make changes to the cluster description.

Reply:
Acknowledged.